pkg/chroottar/chroottar.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chroottar
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	"github.com/coreos/rkt/pkg/multicall"
+	ptar "github.com/coreos/rkt/pkg/tar"
+)
+
+var testing bool
+var chroot = true
+
+const (
+	multicallName = "extracttar"
+	fileMapFdNum  = 3
+)
+
+var mcEntrypoint multicall.Entrypoint
+
+func init() {
+	mcEntrypoint = multicall.Add(multicallName, extractTar)
+}
+
+func SetChroot(enable bool) {
+	chroot = enable
+}
+
+func extractTar() error {
+	if len(os.Args) < 3 {
+		return fmt.Errorf("not enough arguments")
+	}
+	dir := os.Args[1]
+	if !filepath.IsAbs(dir) {
+		return fmt.Errorf("dir %s must be an absolute path", dir)
+	}
+	overwrite, err := strconv.ParseBool(os.Args[2])
+	if err != nil {
+		return fmt.Errorf("error parsing overwrite argument: %v", err)
+	}
+
+	extractDir := "/"
+	if chroot {
```

Comment:
How is this compared to hide this with an interface, e.g.

```go
type DirSetter interface {
    SetDir(dir string)
}
```

Which we can do chroot(dir), chdir("/") in real binary, and do chdir(dir) in unit tests.